showkase/src/main/java/com/airbnb/android/showkase/ui/ShowkaseBrowserApp.kt

```diff
@@ -436,103 +495,151 @@ private fun Map<String, List<*>>.isOnlyCategory(
 ) = this.values.isNotEmpty() && otherCategoryMap1.isEmpty() && otherCategoryMap2.isEmpty()
 
 private fun NavGraphBuilder.componentsNavGraph(
-    navController: NavHostController,
     groupedComponentMap: Map<String, List<ShowkaseBrowserComponent>>,
-    showkaseBrowserScreenMetadata: MutableState<ShowkaseBrowserScreenMetadata>
+    showkaseBrowserScreenMetadata: ShowkaseBrowserScreenMetadata,
+    onRootScreen: Boolean,
+    onUpdateShowkaseBrowserScreenMetadata: (ShowkaseBrowserScreenMetadata) -> Unit,
+    navigateTo: (ShowkaseCurrentScreen) -> Unit,
 ) {
     composable(ShowkaseCurrentScreen.COMPONENT_GROUPS.name) {
         ShowkaseComponentGroupsScreen(
-            groupedComponentMap,
-            showkaseBrowserScreenMetadata,
-            navController
+            groupedComponentMap = groupedComponentMap,
+            showkaseBrowserScreenMetadata = showkaseBrowserScreenMetadata,
+            onUpdateShowkaseBrowserScreenMetadata = onUpdateShowkaseBrowserScreenMetadata,
+            onRootScreen = onRootScreen,
+            navigateTo = navigateTo,
         )
     }
     composable(ShowkaseCurrentScreen.COMPONENTS_IN_A_GROUP.name) {
         ShowkaseComponentsInAGroupScreen(
-            groupedComponentMap,
-            showkaseBrowserScreenMetadata,
-            navController
+            groupedComponentMap = groupedComponentMap,
+            showkaseBrowserScreenMetadata = showkaseBrowserScreenMetadata,
+            onUpdateShowkaseBrowserScreenMetadata = onUpdateShowkaseBrowserScreenMetadata,
+            navigateTo = navigateTo,
         )
     }
     composable(ShowkaseCurrentScreen.COMPONENT_STYLES.name) {
         ShowkaseComponentStylesScreen(
```

Comment:
Similarly, skipping the styles screen when there's only one style would be super nice. Directly to the detail screen.